### ISSUE TITLE
ContainerInfra: Cluster resize error decoding response

### ISSUE THREAD

**Issue by tghartland** [gophercloud/gophercloud#1631]
When using the clusters.Resize method for the new resize API, the cluster is successfully resized but the returned result is always an error with just the message "EOF".

It comes from [here](https://github.com/gophercloud/gophercloud/blob/157432124aab520975efedb5e54b95287785578d/provider_client.go#L479) where the response body is decoded. The reason is that cluster resize does not return a body when the request is successful.

**Comment by tghartland** [comment]
@openstacker can you confirm if the resize response should have a body or not? In #1516 you tested the code and it worked, but now we are running magnum stein and by the CLI client / curl / gophercloud we do not get a body in the resize request, which causes gophercloud to fail as there is nothing to decode.

With `openstack coe cluster resize --debug`:
```
REQ: curl -g -i -X POST https://openstack.cern.ch:9511/v1/clusters/424b2152-c417-40e0-ba75-73748963ebab/actions/resize -H "Accept: application/json" -H "Content-Type: application/json" -H "OpenStack-API-Version: container-infra latest" -H "User-Agent: None" -H "X-Auth-Token: {SHA256}xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx" -d '{"node_count": 6}'
https://openstack.cern.ch:9511 "POST /v1/clusters/424b2152-c417-40e0-ba75-73748963ebab/actions/resize HTTP/1.1" 202 0
RESP: [202] Connection: keep-alive Content-Length: 0 Date: Mon, 24 Jun 2019 08:40:39 GMT OpenStack-API-Maximum-Version: container-infra 1.7 OpenStack-API-Minimum-Version: container-infra 1.1 OpenStack-API-Version: container-infra 1.7 Server: Werkzeug/0.14.1 Python/2.7.5 Strict-Transport-Security: max-age=15768000 vary: OpenStack-API-Version x-openstack-request-id: req-e916c398-d41c-4938-8b45-0ee8eca45d57
RESP BODY: Omitted, Content-Type is set to None. Only application/json responses have their bodies logged.
POST call to container-infra for https://openstack.cern.ch:9511/v1/clusters/424b2152-c417-40e0-ba75-73748963ebab/actions/resize used request id req-e916c398-d41c-4938-8b45-0ee8eca45d57
clean_up ResizeCluster: 
END return value: 0
Request to resize cluster 424b2152-c417-40e0-ba75-73748963ebab has been accepted.
```

I have a fix for gophercloud ready if the empty body response is intended.

**Comment by openstacker** [comment]
@tghartland  There should be a cluster ID being returned. But unfortunately, there is a bug, I have proposed a fix for this, see https://review.opendev.org/672600

**Comment by tghartland** [comment]
@openstacker thanks for clearing that up. Since it is only the ID being returned then in gophercloud there should be an extract method for ResizeResult the same as Create and Update:
https://github.com/gophercloud/gophercloud/blob/00c1d0b31215aa8ee476f9df2fa7f000cfe5130f/openstack/containerinfra/v1/clusters/results.go#L55-L61
I can make that change myself and hopefully have it tested with the magnum patch by next week.

**Comment by jtopjian** [comment]
Implemented in #1649
